### PR TITLE
Distributed tracing 2.9 otel components

### PR DESCRIPTION
**Pull request by pavolloffay**
The preview is at https://github.com/openshift/openshift-docs/pull/63284 

Resolves: 
* https://issues.redhat.com/browse/TRACING-3308
* https://issues.redhat.com/browse/TRACING-3263
* https://issues.redhat.com/browse/TRACING-3265
* https://issues.redhat.com/browse/TRACING-3267
* https://issues.redhat.com/browse/TRACING-3268
* https://issues.redhat.com/browse/TRACING-3269
* https://issues.redhat.com/browse/TRACING-3266

### CODE REVIEW

**Comment by pavolloffay** [comment]
@max-cx could you please review?

**Review by iblancasa — APPROVED** [review]

File: modules/distr-tracing-config-otel-collector.adoc [hunk 1]
```diff
@@ -162,3 +162,214 @@ If a component is configured, but not defined within the `service` section then
...
 |None
 |===
+
+== OpenTelemetry Collector components
+
+=== Receivers
+
+==== OTLP Receiver
+
+OTLP receiver ingests data in the OpenTelemetry protocol (OTLP).
```
Comment (max-cx) [inline modules/distr-tracing-config-otel-collector.adoc]:
```suggestion
OTLP receiver ingests data using the OpenTelemetry protocol (OTLP).
```

File: modules/distr-tracing-config-otel-collector.adoc [hunk 2]
```diff
@@ -162,3 +162,214 @@ If a component is configured, but not defined within the `service` section then
...
+
+* Support level: tech-preview
```
Comment (max-cx) [inline modules/distr-tracing-config-otel-collector.adoc]:
```suggestion
* Support level: link:https://access.redhat.com/support/offerings/techpreview[Technology Preview]
```

File: modules/distr-tracing-config-otel-collector.adoc [hunk 3]
```diff
@@ -162,3 +162,214 @@ If a component is configured, but not defined within the `service` section then
...
+    service:
+      pipelines:
+        traces:
+          receivers: [otlp]
+        metrics:
+          receivers: [otlp]
+----
+<1> The OTLP gRPC endpoint. If omitted the default `+0.0.0.0:4317+` is used.
+<2> The OTLP HTTP endpoint . If omitted the default `+0.0.0.0:4318+` is used.
+<3> The TLS server side configuration. Defines paths to TLS certificates. If omitted TLS is disabled.
```
Comment (max-cx) [inline modules/distr-tracing-config-otel-collector.adoc]:
```suggestion
.OpenTelemetry collector custom resource with enabled OTLP receiver
====
[source,yaml]
----
  config: |
    receivers:
      otlp:
        protocols:
          grpc:
            endpoint: 0.0.0.0:4317 <1>
          http:
            endpoint: 0.0.0.0:4318 <2>
          tls: <3>
            ca_file: ca.pem
            cert_file: cert.pem
            key_file: key.pem

    service:
      pipelines:
        traces:
          receivers: [otlp]
        metrics:
          receivers: [otlp]
----
<1> The OTLP gRPC endpoint. If omitted, the default `+0.0.0.0:4317+` is used.
<2> The OTLP HTTP endpoint . If omitted, the default `+0.0.0.0:4318+` is used.
<3> The TLS server side configuration. Defines paths to TLS certificates. If omitted, TLS is disabled.
====
```

File: modules/distr-tracing-config-otel-collector.adoc [hunk 4]
```diff
@@ -162,3 +162,214 @@ If a component is configured, but not defined within the `service` section then
...
+
+==== Jaeger Receiver
+
+Jaeger receiver ingests data in Jaeger formats.
+
+* Support level: tech-preview
```
Comment (max-cx) [inline modules/distr-tracing-config-otel-collector.adoc]:
```suggestion
* Support level: link:https://access.redhat.com/support/offerings/techpreview[Technology Preview]
```

File: modules/distr-tracing-config-otel-collector.adoc [hunk 5]
```diff
@@ -162,3 +162,214 @@ If a component is configured, but not defined within the `service` section then
...
+    service:
+      pipelines:
+        traces:
+          receivers: [jaeger]
+----
+<1> The Jaeger gRPC endpoint. If omitted the default `+0.0.0.0:14250+` is used.
+<2> The Jaeger Thrift HTTP endpoint. If omitted the default `+0.0.0.0:14268+` is used.
+<3> The Jaeger Thrift Compact endpoint. If omitted the default `+0.0.0.0:6831+` is used.
+<4> The Jaeger Thrift Binary endpoint. If omitted the default `+0.0.0.0:6832+` is used.
+<5> The TLS server side configuration. More details are provided in the OTLP receiver configuration section.
```
Comment (max-cx) [inline modules/distr-tracing-config-otel-collector.adoc]:
```suggestion
.OpenTelemetry collector custom resource with enabled Jaeger receiver
====
[source,yaml]
----
  config: |
    receivers:
      jaeger:
        protocols:
          grpc:
            endpoint: 0.0.0.0:14250 <1>
          thrift_http:
            endpoint: 0.0.0.0:14268 <2>
          thrift_compact:
            endpoint: 0.0.0.0:6831 <3>
          thrift_binary:
            endpoint: 0.0.0.0:6832 <4>
          tls: <5>

    service:
      pipelines:
        traces:
          receivers: [jaeger]
----
<1> The Jaeger gRPC endpoint. If omitted, the default `+0.0.0.0:14250+` is used.
<2> The Jaeger Thrift HTTP endpoint. If omitted, the default `+0.0.0.0:14268+` is used.
<3> The Jaeger Thrift Compact endpoint. If omitted, the default `+0.0.0.0:6831+` is used.
<4> The Jaeger Thrift Binary endpoint. If omitted, the default `+0.0.0.0:6832+` is used.
<5> The TLS server side configuration. See the OTLP receiver configuration section for more details.
====
```

File: modules/distr-tracing-config-otel-collector.adoc [hunk 6]
```diff
@@ -162,3 +162,214 @@ If a component is configured, but not defined within the `service` section then
...
+
+==== Zipkin Receiver
+
+Zipkin receiver ingests data in Zipkin V1 and V2 formats.
+
+* Support level: tech-preview
```
Comment (max-cx) [inline modules/distr-tracing-config-otel-collector.adoc]:
```suggestion
* Support level: link:https://access.redhat.com/support/offerings/techpreview[Technology Preview]
```

File: modules/distr-tracing-config-otel-collector.adoc [hunk 7]
```diff
@@ -162,3 +162,214 @@ If a component is configured, but not defined within the `service` section then
...
+        endpoint: 0.0.0.0:9411 <1>
+        tls: <2>
+
+    service:
+      pipelines:
+        traces:
+          receivers: [zipkin]
+----
+<1> The Zipkin HTTP endpoint. If omitted the default `+0.0.0.0:9411+` is used.
+<2> The TLS server side configuration. More details are provided in the OTLP receiver configuration section.
```
Comment (max-cx) [inline modules/distr-tracing-config-otel-collector.adoc]:
```suggestion
.OpenTelemetry collector custom resource with enabled Zipkin receiver
====
[source,yaml]
----
  config: |
    receivers:
      zipkin:
        endpoint: 0.0.0.0:9411 <1>
        tls: <2>

    service:
      pipelines:
        traces:
          receivers: [zipkin]
----
<1> The Zipkin HTTP endpoint. If omitted, the default `+0.0.0.0:9411+` is used.
<2> The TLS server side configuration. See the OTLP receiver configuration section for more details.
====
```

File: modules/distr-tracing-config-otel-collector.adoc [hunk 8]
```diff
@@ -162,3 +162,214 @@ If a component is configured, but not defined within the `service` section then
...
+----
+<1> The Jaeger gRPC endpoint. If omitted the default `+0.0.0.0:14250+` is used.
+<2> The Jaeger Thrift HTTP endpoint. If omitted the default `+0.0.0.0:14268+` is used.
+<3> The Jaeger Thrift Compact endpoint. If omitted the default `+0.0.0.0:6831+` is used.
+<4> The Jaeger Thrift Binary endpoint. If omitted the default `+0.0.0.0:6832+` is used.
+<5> The TLS server side configuration. More details are provided in the OTLP receiver configuration section.
+
+==== Zipkin Receiver
+
+Zipkin receiver ingests data in Zipkin V1 and V2 formats.
```
Comment (max-cx) [inline modules/distr-tracing-config-otel-collector.adoc]:
```suggestion
Zipkin receiver ingests data in the Zipkin V1 and V2 formats.
```

File: modules/distr-tracing-config-otel-collector.adoc [hunk 9]
```diff
@@ -162,3 +162,214 @@ If a component is configured, but not defined within the `service` section then
...
+<1> The Zipkin HTTP endpoint. If omitted the default `+0.0.0.0:9411+` is used.
+<2> The TLS server side configuration. More details are provided in the OTLP receiver configuration section.
+
+=== Exporters
+
+==== OTLP Exporter
+
+OTLP gRPC exporter exports data in the OpenTelemetry protocol (OTLP).
+
+* Support level: tech-preview
```
Comment (max-cx) [inline modules/distr-tracing-config-otel-collector.adoc]:
```suggestion
* Support level: link:https://access.redhat.com/support/offerings/techpreview[Technology Preview]
```

File: modules/distr-tracing-config-otel-collector.adoc [hunk 10]
```diff
@@ -162,3 +162,214 @@ If a component is configured, but not defined within the `service` section then
...
+----
+<1> The OTLP gRPC endpoint. If `+https://+` scheme is used then client transport security is enabled and overrides `+insecure+` setting in `+tls+`.
+<2> The client side TLS configuration. Defines paths to TLS certificates.
+<3> Headers are sent for every RPC performed during an established connection.
+
+==== OTLP HTTP Exporter
+
+OTLP HTTP exporter exports data in the OpenTelemetry protocol (OTLP).
+
+* Support level: tech-preview
```
Comment (max-cx) [inline modules/distr-tracing-config-otel-collector.adoc]:
```suggestion
* Support level: link:https://access.redhat.com/support/offerings/techpreview[Technology Preview]
```

File: modules/distr-tracing-config-otel-collector.adoc [hunk 11]
```diff
@@ -162,3 +162,214 @@ If a component is configured, but not defined within the `service` section then
...
+----
+<1> The OTLP HTTP endpoint. If `+https://+` scheme is used then client transport security is enabled and overrides `+insecure+` setting in `+tls+`.
+<2> The client side TLS configuration. Defines paths to TLS certificates.
+<3> Headers are sent in every HTTP request.
+
+==== Jaeger Exporter
+
+Jaeger exporter exports data in the Jaeger proto format through gRPC.
+
+* Support level: tech-preview
```
Comment (max-cx) [inline modules/distr-tracing-config-otel-collector.adoc]:
```suggestion
* Support level: link:https://access.redhat.com/support/offerings/techpreview[Technology Preview]
```

File: modules/distr-tracing-config-otel-collector.adoc [hunk 12]
```diff
@@ -162,3 +162,214 @@ If a component is configured, but not defined within the `service` section then
...
+          exporters: [jaeger]
+----
+<1> The Jaeger gRPC endpoint.
+<2> The client side TLS configuration. Defines paths to TLS certificates.
+
+==== Logging exporter
+
+Logging exporter prints data the standard output.
+
+* Support level: tech-preview
```
Comment (max-cx) [inline modules/distr-tracing-config-otel-collector.adoc]:
```suggestion
* Support level: link:https://access.redhat.com/support/offerings/techpreview[Technology Preview]
```

File: modules/distr-tracing-config-otel-collector.adoc [hunk 13]
```diff
@@ -162,3 +162,214 @@ If a component is configured, but not defined within the `service` section then
...
+      logging:
+        verbosity: detailed <1>
+    service:
+      pipelines:
+        traces:
+          exporters: [logging]
+        metrics:
+          exporters: [logging]
+----
+<1> The verbosity of the logging export (`+detailed+`|`+normal+`|`+basic+`). When set to detailed, pipeline data is verbosely logged. Defaults to `+normal+`.
```
Comment (max-cx) [inline modules/distr-tracing-config-otel-collector.adoc]:
```suggestion
.OpenTelemetry collector custom resource with enabled Logging exporter
====
[source,yaml]
----
  config: |
    exporters:
      logging:
        verbosity: detailed <1>
    service:
      pipelines:
        traces:
          exporters: [logging]
        metrics:
          exporters: [logging]
----
<1> The verbosity of the logging export (`+detailed+`|`+normal+`|`+basic+`). When set to detailed, pipeline data is verbosely logged. Defaults to `+normal+`.
====
```

File: modules/distr-tracing-config-otel-collector.adoc [hunk 14]
```diff
@@ -162,3 +162,214 @@ If a component is configured, but not defined within the `service` section then
...
+      pipelines:
+        traces:
+          exporters: [jaeger]
+----
+<1> The Jaeger gRPC endpoint.
+<2> The client side TLS configuration. Defines paths to TLS certificates.
+
+==== Logging exporter
+
+Logging exporter prints data the standard output.
```
Comment (max-cx) [inline modules/distr-tracing-config-otel-collector.adoc]:
```suggestion
Logging exporter prints data to the standard output.
```

File: modules/distr-tracing-config-otel-collector.adoc [hunk 15]
```diff
@@ -162,3 +162,214 @@ If a component is configured, but not defined within the `service` section then
...
+          ca_file: ca.pem
+          cert_file: cert.pem
+          key_file: key.pem
+    service:
+      pipelines:
+        traces:
+          exporters: [jaeger]
+----
+<1> The Jaeger gRPC endpoint.
+<2> The client side TLS configuration. Defines paths to TLS certificates.
```
Comment (max-cx) [inline modules/distr-tracing-config-otel-collector.adoc]:
```suggestion
.OpenTelemetry collector custom resource with enabled Jaeger exporter
====
[source,yaml]
----
  config: |
    exporters:
      jaeger:
        endpoint: jaeger-all-in-one:14250 <1>
        tls: <2>
          ca_file: ca.pem
          cert_file: cert.pem
          key_file: key.pem
    service:
      pipelines:
        traces:
          exporters: [jaeger]
----
<1> The Jaeger gRPC endpoint.
<2> The client side TLS configuration. Defines paths to TLS certificates.
====
```

File: modules/distr-tracing-config-otel-collector.adoc [hunk 16]
```diff
@@ -162,3 +162,214 @@ If a component is configured, but not defined within the `service` section then
...
+    service:
+      pipelines:
+        traces:
+          exporters: [otlphttp]
+        metrics:
+          expoters: [otlphttp]
+----
+<1> The OTLP HTTP endpoint. If `+https://+` scheme is used then client transport security is enabled and overrides `+insecure+` setting in `+tls+`.
+<2> The client side TLS configuration. Defines paths to TLS certificates.
+<3> Headers are sent in every HTTP request.
```
Comment (max-cx) [inline modules/distr-tracing-config-otel-collector.adoc]:
```suggestion
.OpenTelemetry collector custom resource with enabled OTLP exporter
====
[source,yaml]
----
  config: |
    exporters:
      otlphttp:
        endpoint: http://tempo-ingester:4318 <1>
        tls: <2>
          ca_file: ca.pem
          cert_file: cert.pem
          key_file: key.pem
        headers: <3>
          X-Scope-OrgID: "dev"

    service:
      pipelines:
        traces:
          exporters: [otlphttp]
        metrics:
          expoters: [otlphttp]
----
<1> The OTLP HTTP endpoint. If `+https://+` scheme is used, then client transport security is enabled and overrides `+insecure+` setting in `+tls+`.
<2> The client side TLS configuration. Defines paths to TLS certificates.
<3> Headers are sent in every HTTP request.
====
```

File: modules/distr-tracing-config-otel-collector.adoc [hunk 17]
```diff
@@ -162,3 +162,214 @@ If a component is configured, but not defined within the `service` section then
...
+        metrics:
+          expoters: [otlp]
+----
+<1> The OTLP gRPC endpoint. If `+https://+` scheme is used then client transport security is enabled and overrides `+insecure+` setting in `+tls+`.
+<2> The client side TLS configuration. Defines paths to TLS certificates.
+<3> Headers are sent for every RPC performed during an established connection.
+
+==== OTLP HTTP Exporter
+
+OTLP HTTP exporter exports data in the OpenTelemetry protocol (OTLP).
```
Comment (max-cx) [inline modules/distr-tracing-config-otel-collector.adoc]:
```suggestion
OTLP HTTP exporter exports data using the OpenTelemetry protocol (OTLP).
```

File: modules/distr-tracing-config-otel-collector.adoc [hunk 18]
```diff
@@ -162,3 +162,214 @@ If a component is configured, but not defined within the `service` section then
...
+    service:
+      pipelines:
+        traces:
+          exporters: [otlp]
+        metrics:
+          expoters: [otlp]
+----
+<1> The OTLP gRPC endpoint. If `+https://+` scheme is used then client transport security is enabled and overrides `+insecure+` setting in `+tls+`.
+<2> The client side TLS configuration. Defines paths to TLS certificates.
+<3> Headers are sent for every RPC performed during an established connection.
```
Comment (max-cx) [inline modules/distr-tracing-config-otel-collector.adoc]:
```suggestion
.OpenTelemetry collector custom resource with enabled OTLP exporter
====
[source,yaml]
----
  config: |
    exporters:
      otlp:
        endpoint: tempo-ingester:4317 <1>
        tls: <2>
          ca_file: ca.pem
          cert_file: cert.pem
          key_file: key.pem
        headers: <3>
          X-Scope-OrgID: "dev"
    service:
      pipelines:
        traces:
          exporters: [otlp]
        metrics:
          expoters: [otlp]
----
<1> The OTLP gRPC endpoint. If `+https://+` scheme is used, then client transport security is enabled and overrides `+insecure+` setting in `+tls+`.
<2> The client side TLS configuration. Defines paths to TLS certificates.
<3> Headers are sent for every RPC performed during an established connection.
====
```

File: modules/distr-tracing-config-otel-collector.adoc [hunk 19]
```diff
@@ -162,3 +162,214 @@ If a component is configured, but not defined within the `service` section then
...
+          receivers: [zipkin]
+----
+<1> The Zipkin HTTP endpoint. If omitted the default `+0.0.0.0:9411+` is used.
+<2> The TLS server side configuration. More details are provided in the OTLP receiver configuration section.
+
+=== Exporters
+
+==== OTLP Exporter
+
+OTLP gRPC exporter exports data in the OpenTelemetry protocol (OTLP).
```
Comment (max-cx) [inline modules/distr-tracing-config-otel-collector.adoc]:
```suggestion
OTLP gRPC exporter exports data using the OpenTelemetry protocol (OTLP).
```

**Pull request merged**